src/overlays/actors/ovl_Boss_Goma/z_boss_goma.c

```diff
@@ -21,115 +111,1836 @@ const ActorInit Boss_Goma_InitVars = {
+                this->patienceTimer = 200;
+                func_80064534(globalCtx, &globalCtx->csCtx);
+                func_8002DF54(globalCtx, &this->actor, 7);
+            }
+            break;
+    }
+
+    if (this->subCameraId != 0) {
+        func_800C04D8(globalCtx, this->subCameraId, &this->subCameraAt, &this->subCameraEye);
+    }
+}
+
+void BossGoma_Defeated(BossGoma* this, GlobalContext* globalCtx) {
+    static Vec3f _vel1 = { 0, 0, 0 };
+    static Vec3f _accel1 = { 0, 1, 0 };
+    static Color_RGBA8_n _color1 = { 255, 255, 255, 255 };
+    static Color_RGBA8_n _color2 = { 0, 100, 255, 255 };
+    static Vec3f _vel2 = { 0, 0, 0 };
+    static Vec3f _accel2 = { 0, -0.5f, 0 };
+    static Vec3f roomCenter = { -150, 0, -350 };
+
+    f32 dx;
+    f32 dz;
+    s16 j;
+    Vec3f vel1 = _vel1;
+    Vec3f accel1 = _accel1;
+    Color_RGBA8_n color1 = _color1;
+    Color_RGBA8_n color2 = _color2;
+    Vec3f vel2 = _vel2;
+    Vec3f accel2 = _accel2;
+    Vec3f pos;
+    Camera* camera;
+    Player* player = PLAYER;
+    Vec3f childPos;
+    s16 i;
+
+    SkelAnime_FrameUpdateMatrix(&this->skelanime);
+    Math_SmoothScaleMaxS(&this->actor.shape.rot.x, 0, 2, 0xBB8);
+
+    if (func_800A56C8(&this->skelanime, 107.0f)) {
+        BossGoma_PlayEffectsAndSfx(this, globalCtx, 0, 8);
+        func_800A9F6C(0.0f, 0x96, 0x14, 0x14);
+    }
+
+    this->visualState = VISUALSTATE_DEFEATED;
+    this->eyeState = EYESTATE_IRIS_NO_FOLLOW_NO_IFRAMES;
+
+    // timer setup to 1200
+    if (this->framesUntilNextAction == 1001) {
+        for (i = 0; i < 90; i++) {
+            if (D_8091B244[i] != 0) {
+                this->deadLimbsState[i] = 1;
+            }
+        }
+    }
+
+    if (this->framesUntilNextAction < 1200 && this->framesUntilNextAction > 1100 &&
+        this->framesUntilNextAction % 8 == 0) {
+        func_800299AC(globalCtx, &this->actor.posRot2.pos);
+    }
+
+    if (this->framesUntilNextAction < 1080 && this->actionState < 3) {
+        if (this->framesUntilNextAction < 1070) {
+            Audio_PlayActorSound2(&this->actor, 0x301C);
+        }
+        for (i = 0; i < 4; i++) {
+            // @bug this 0-indexes into this->defeatedLimbPositions which is initialized with
+            // this->defeatedLimbPositions[limb] but limb is 1-indexed in skelanime callbacks, this means effects
+            // should spawn at this->defeatedLimbPositions[0] too, which is uninitialized, so map origin?
+            j = (s16)(Math_Rand_ZeroOne() * 85);
+            if (this->defeatedLimbPositions[j].y < 10000.0f) {
+                pos.x = Math_Rand_CenteredFloat(20.0f) + this->defeatedLimbPositions[j].x;
+                pos.y = Math_Rand_CenteredFloat(10.0f) + this->defeatedLimbPositions[j].y;
+                pos.z = Math_Rand_CenteredFloat(20.0f) + this->defeatedLimbPositions[j].z;
+                func_8002836C(globalCtx, &pos, &vel1, &accel1, &color1, &color2, 0x1F4, 0xA, 0xA);
+            }
```

Comment:
after you merge effects, most args will make sense in decimal. this goes for any effects in the file